coderd/database/testcases/queue_position.go

```diff
@@ -0,0 +1,245 @@
+package testcases
```

Comment:
I think this is an interesting approach, but I'm not a fan of the disconnect between assumptions of how tests need to be set up and the actual test data. And I guess I'm not really sold on testing the same cases in both places. IMO testing in `coderd` only is fine, and if the query is sufficiently complex, testing the edge cases in `database/querier_test.go` is also fine. But perhaps I'm just not seeing the value? Happy to be enlightened if that's the case.